ta4j-core/src/main/java/org/ta4j/core/indicators/helpers/DifferencePercentage.java

```diff
@@ -0,0 +1,78 @@
+/*
+  The MIT License (MIT)
+
+  Copyright (c) 2014-2017 Marc de Verdelhan, Ta4j Organization & respective authors (see AUTHORS)
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy of
+  this software and associated documentation files (the "Software"), to deal in
+  the Software without restriction, including without limitation the rights to
+  use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+  the Software, and to permit persons to whom the Software is furnished to do so,
+  subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in all
+  copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+  FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+  COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+  IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.helpers;
+
+import org.ta4j.core.Indicator;
+import org.ta4j.core.indicators.CachedIndicator;
+import org.ta4j.core.num.Num;
+
+/**
+ * Difference Change Indicator.
+ * <p>
+ * Get the difference in percentage from the last time the threshold was reached.
+ * <p>
+ * Or if you don't pass the threshold you will always just get the difference from the precious value.
+ * </p>
+ */
+public class DifferencePercentage extends CachedIndicator<Num> {
+    private Indicator<Num> indicator;
+    private final Num percentageThreshold;
+    private Num lastNotification;
+
+    public DifferencePercentage(Indicator<Num> indicator) {
+        this(indicator, indicator.numOf(0));
+    }
+
+    public DifferencePercentage(Indicator<Num> indicator, Number percentageThreshold) {
+        this(indicator, indicator.numOf(percentageThreshold));
+    }
+
+    public DifferencePercentage(Indicator<Num> indicator, Num percentageThreshold) {
+        super(indicator);
+
+        this.indicator = indicator;
+        this.percentageThreshold = percentageThreshold;
+    }
+
+    @Override
+    protected Num calculate(int index) {
+        Num value = indicator.getValue((index));
+        if (lastNotification == null) {
+            lastNotification = value;
+            return numOf(0);
```

Comment:
If there is no previous value, the Indicator cannot return a valid result. In such cases ta4js convention is to return `NaN` to avoid incorrect interpretation of the result

Reply:
Thanks, didn't know about that, much better :)